Core/Dialog/SavedataParam.cpp

```diff
@@ -382,6 +385,40 @@ int SavedataParam::DeleteData(SceUtilitySavedataParam* param) {
 	ClearCaches();
 	pspFileSystem.RemoveFile(filePath);
 
+	// Update PARAM.SFO to remove the file, if it was ni the list.
+	std::shared_ptr<ParamSFOData> sfoFile = LoadCachedSFO(sfoPath);
```

Comment:
this could be a `unique_ptr`, right? (less overhead, though hardly matters here)

Reply:
`LoadCachedSFO()` returns a shared_ptr, because it returns a common pointer from a cache but you can clear its cache, so I just wanted to be safe from any crashes.  For example, saving a file loads the SFO data, then clears the SFO cache (because it modifies things.)

It definitely can't be a unique_ptr, because the whole point of `LoadCachedSFO()` is to reuse the already loaded SFO data.  Would not want it freed on each use.

-[Unknown]